api/controllers/apps_test.go

```diff
@@ -74,9 +83,11 @@ func TestAppCreate(t *testing.T) {
 	}
 }
 
-func TestAppCreateWithAlreadyExists(t *testing.T) {
+func TestAppCreateWithAlreadyExistsUnbound(t *testing.T) {
 	aws := test.StubAws(
+		test.DescribeAppStackCycle("application"),
 		test.CreateAppStackExistsCycle("application"),
```

Comment:
Ideally this test case can return a 403 based on a single CF call:

```
test.DescribeAppStackCycle("application")
```

It would also make the test clearer to demonstrate that the error string says something about a legacy app.